pkg/api/v1beta3/dynakube/activegate/spec.go

```diff
@@ -64,7 +64,7 @@ type dependencies struct {
 }
 
 func (d dependencies) Any() bool {
-	return d.extensions
+	return d.extensions // kspm is a dependency too, but blocked by validation webhook to not run standalone
```

Comment:
you can even remove the `kspm` dependency as it is no longer used/needed (the creation of the Service was the only reason for it to exist)

Reply:
I don't quite get what you mean - As far as I got it, KSPM requires an ActiveGate and to communicate with the ActiveGate it needs a service. Why shouldn't there be a dependency and what would you remove? - The unused `kspm` variable?

Reply:
KSPM does need an activeGate, but it has to be explicitly defined in the `DynaKube`, otherwise our validation will complain.

This `kspm` dependency was introduced only to create the `Service` (if you look at where it's used, you will see 😉 ).
- This was needed because the "required" activeGate capability was `kubernetes-monitoring`, this didn't result in the creation of a `Service`
- but the node-collector wanted to send stuff to the activeGate, so the `Service` was needed
- at that point we didn't know if it's ok to "just create the Service" every time, so this hacky solution was added.

Reply:
Isn't that exactly what I explained above in Adams question 😅 

That's why I created the comment, that the dependency is actually not required. - It's still a dependency though and we just rely on the validation webhook to block if the ActiveGate capability is not there. That's why I would leave the comment in there.

Reply:
but you don't need the `bool` :D (you can leave the comment if you want)